index.js

```diff
@@ -15,6 +15,46 @@ var NodeWebkitBrowser = function(baseBrowserDecorator, args,config) {
   });
   searchPaths.unshift(process.env.NODE_PATH);
 
+  this._findCommand = function(callback) {
+    var command = this._getCommand();
+
+    if (command !== null) {
+      callback(null, command);
+
+      return;
+    }
+
+    /*
+     * Assuming this script is at <project_root>/node_modules/karma-nwjs-phoenix-launcher,
+     * this should resolve to <project_root>/package.json.
+     *
+     * It won't work if the package is installed via `npm link` because in that scenario
+     * `__dirname` provides the module's original path, not its installed path.
+     */
+    var pkgPath = path.join(path.resolve(__dirname, "../.."), "package.json");
+    var pkg = JSON.parse(fs.readFileSync(pkgPath));
+    var nwVersion = pkg.build.nwVersion;
+  
+    (new builder.Downloader({
+      platform: process.platform,
+      arch: process.arch,
+      version: nwVersion,
+      flavor: 'sdk',
+      useCaches: true,
+      showProgress: true
+    }))
+    .fetchAndExtract()
+    .then(function(extracted) {
+      return builder.findExecutable(process.platform, extracted);
+    })
+    .then(function(found) {
+      callback(null, found);
+    }, function(err) {
```

Comment:
IMO using the `catch` ability from [Promises](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Promise) is more readable than the second arg to `then`

Reply:
Actually, that catches a bit more in this instance, so is the correct thing to do. Sometimes it isn't, so I have a habit of using the second arg.